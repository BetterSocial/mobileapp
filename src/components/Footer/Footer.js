import * as React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import MemoIcCreatePoll from '../../assets/icons/ic_create_poll';
import MemoIc_arrow_down_vote_off from '../../assets/arrow/Ic_downvote_off';
import MemoIc_arrow_down_vote_on from '../../assets/arrow/Ic_downvote_on';
import MemoIc_arrow_upvote_off from '../../assets/arrow/Ic_upvote_off';
import MemoIc_arrow_upvote_on from '../../assets/arrow/Ic_upvote_on';
import MemoIc_block_inactive from '../../assets/block/Ic_block_inactive';
import MemoIc_comment from '../../assets/icons/Ic_comment';
import MemoIc_share from '../../assets/icons/Ic_share';
import MemoPollWinnerBadge from '../../assets/icon/IconPollWinnerBadge';
import { ENABLE_DEV_ONLY_FEATURE } from '../../utils/constants';
import {FONTS, SIZES} from '../../utils/theme';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const Footer = ({
  onPressShare,
  onPressComment,
  totalComment,
  isSelf,
  onPressBlock,
  statusVote = 'none',
  onPressDownVote,
  onPressUpvote,
  onPressScore,
  totalVote,
  disableComment = false,
  blockStatus,
  loadingVote,
  showScoreButton = false,
}) => {
  const handleBlockUi = () => {
    if (isSelf) {
      return null;
    } 
      if (blockStatus && blockStatus.blocker) {
        return null;
      } 
        return (
          <TouchableOpacity testID='onPressBlock' style={styles.btn} onPress={onPressBlock}>
            <View style={styles.btnBlock}>
              <MemoIc_block_inactive height={22} width={22} />
            </View>
          </TouchableOpacity>
        );
      
    
  };

  // const __renderShowScoreUI = () => {
  //   if(false && showScoreButton) {
  //     return <TouchableOpacity style={styles.btn} onPress={onPressScore}>
  //       <View style={styles.btnComment}>
  //         <MemoIcCreatePoll height={24} width={25}/>
  //       </View>
  //     </TouchableOpacity>
  //   } return <></>
  // }

  return (
    <View style={[styles.rowSpaceBeetwen, styles.container]}>
      <View style={styles.leftGroupContainer}>
        <TouchableOpacity style={styles.btn} onPress={onPressShare}>
          <View style={styles.btnShare}>
            <MemoIc_share height={20} width={22} />
          </View>
        </TouchableOpacity>
        {disableComment ? (
          <View style={styles.btn}>
            <View style={styles.btnComment}>
              <MemoIc_comment height={24} width={25} />
            </View>
          </View>
        ) : (
          <TouchableOpacity style={styles.btn} onPress={onPressComment}>
            <View style={styles.btnComment}>
              <MemoIc_comment height={24} width={25} />
            </View>
          </TouchableOpacity>
        )}
      </View>
      {disableComment ? (
        <View style={styles.totalCommentContainer}>
          <Text style={styles.text}>{totalComment}</Text>
        </View>
      ) : (
        <TouchableOpacity style={{ flex: 1 }} onPress={onPressComment}>
          <View style={styles.totalCommentContainer}>
            <Text style={styles.text}>{totalComment}</Text>
          </View>
        </TouchableOpacity>
      )}
      {/* {__renderShowScoreUI()} */}
      <View style={styles.rightGroupContainer}>
        {handleBlockUi()}
        <TouchableOpacity
          testID='downVoteBtn'
          disabled={loadingVote}
          style={styles.btn}
          onPress={onPressDownVote}>
          <View style={styles.btnDownvote}>
            {statusVote === 'downvote' ? (
              <MemoIc_arrow_down_vote_on width={20} height={18} />
            ) : (
              <MemoIc_arrow_down_vote_off width={20} height={18} />
            )}
          </View>
        </TouchableOpacity>

        <Text style={styles.vote(totalVote)}>{totalVote}</Text>

        <TouchableOpacity
          disabled={loadingVote}
          style={styles.btn}
          onPress={onPressUpvote}>
          <View style={styles.btnUpvote}>
            {statusVote === 'upvote' ? (
              <MemoIc_arrow_upvote_on width={20} height={18} />
            ) : (
              <MemoIc_arrow_upvote_off width={20} height={18} />
            )}
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    // height: '100%',
  },
  leftGroupContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    // marginLeft: 8,
  },
  rightGroupContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    // marginRight: 8,
  },
  rowSpaceBeetwen: {
    flexDirection: 'row',
    alignItems: 'center',
    // justifyContent: 'space-between',
    // marginHorizontal: 10,
  },
  text: {
    ...FONTS.body3,
    color: '#C4C4C4',
  },
  vote: (count) => ({
    ...FONTS.body3,
    textAlign: 'center',
    width: 26,
    color: count > 0 ? '#00ADB5' : count < 0 ? '#FF2E63' : '#C4C4C4',
  }),
  btn: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
  },
  btnShare: {
    height: '100%',
    justifyContent: 'center',
    paddingRight: 13.5,
    paddingLeft: 18,
    // backgroundColor: 'red',
  },
  btnComment: {
    height: '100%',
    justifyContent: 'center',
    paddingLeft: 13.5,
    paddingRight: 10,
    // backgroundColor: 'blue',
  },
  btnBlock: {
    height: '100%',
    justifyContent: 'center',
    paddingHorizontal: 10,
    // backgroundColor: 'gray',
  },
  btnDownvote: {
    height: '100%',
    justifyContent: 'center',
    paddingHorizontal: 11,
    // backgroundColor: 'green',
  },
  btnUpvote: {
    height: '100%',
    justifyContent: 'center',
    paddingLeft: 11,
    paddingRight: 22,
    // backgroundColor: 'brown',
  },
  totalCommentContainer: {
    flex: 1,
    // backgroundColor: 'blue',
    height: '100%',
    justifyContent: 'center',
  },
  buttonFollowing: {
    width: 88,
    height: 36,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: colors.bondi_blue,
    borderRadius: 8,
  },
  textButtonFollowing: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 12,
    color: colors.bondi_blue,
  },
});

Footer.propTypes = {
  onPressShare: PropTypes.func,
  onPressComment: PropTypes.func,
  onPressBlock: PropTypes.func,
  onPressUpvote: PropTypes.func,
  onPressDownVote: PropTypes.func,
  item: PropTypes.object.isRequired,
  totalVote: PropTypes.number,
  totalComment: PropTypes.number,
  statusVote: PropTypes.oneOf(['none', 'upvote', 'downvote']),
  isSelf: PropTypes.bool,
};

export default Footer;
