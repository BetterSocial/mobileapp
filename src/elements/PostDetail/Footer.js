import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import IconIc from 'react-native-vector-icons/Ionicons';
import IconEn from 'react-native-vector-icons/Entypo';
import IconMt from 'react-native-vector-icons/MaterialIcons';
import IconMtC from 'react-native-vector-icons/MaterialCommunityIcons';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
const Footer = ({onBlock}) => {
  return (
    <View style={styles.constainer}>
      <TouchableOpacity style={styles.btn}>
        <IconIc
          name="ios-share-social-outline"
          size={17}
          color={colors.gray1}
        />
      </TouchableOpacity>
      <TouchableOpacity style={styles.btn} onPress={() => onBlock()}>
        <IconEn name="block" size={17} color={colors.gray1} />
      </TouchableOpacity>
      <TouchableOpacity style={styles.btn}>
        <IconMt name="chat" size={17} color={colors.gray1} />
      </TouchableOpacity>
      <TouchableOpacity style={styles.arrowup}>
        <IconMtC name="arrow-up-bold-outline" size={17} color={colors.gray1} />
      </TouchableOpacity>
      <Text style={styles.count}>0</Text>
      <TouchableOpacity>
        <IconMtC
          name="arrow-down-bold-outline"
          size={17}
          color={colors.gray1}
        />
      </TouchableOpacity>
    </View>
  );
};

export default Footer;

const styles = StyleSheet.create({
  constainer: {
    marginHorizontal: 22,
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  btn: {
    marginHorizontal: 10,
  },
  count: {
    fontFamily: fonts.inter[400],
    fontSize: 14,
    marginHorizontal: 13,
    color: colors.gray1,
  },
  arrowup: {
    marginLeft: 20,
  },
});
