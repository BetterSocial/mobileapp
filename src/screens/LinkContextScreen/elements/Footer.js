import * as React from 'react';
import {StyleSheet, View} from 'react-native';

import {COLORS} from '../../../utils/theme';
import {Footer, Gap, PreviewComment} from '../../../components';
import {getCountCommentWithChild, getCountVote} from '../../../utils/getstream';

const LinkContextScreenFooter = ({
  item,
  itemId,
  onPressBlock,
  onPressShare,
  onPressComment,
  onPressDownVote,
  onPressUpvote,
  selfUserId
}) => {
  const [previewComment, setPreviewComment] = React.useState({});
  const [isReaction, setReaction] = React.useState(false);
  const [voteStatus, setVoteStatus] = React.useState('none');
  const [statusUpvote, setStatusUpvote] = React.useState(false);
  const [statusDownvote, setStatusDowvote] = React.useState(false);
  const [totalVote, setTotalVote] = React.useState(0);
  //   const [item, setItem] = React.useState(postItem);

  React.useEffect(() => {
    const validationStatusVote = () => {
      if (item.reaction_counts !== undefined || null) {
        if (item.latest_reactions.upvotes !== undefined) {
          const upvote = item.latest_reactions.upvotes.filter(
            (vote) => vote.user_id === selfUserId
          );
          if (upvote !== undefined) {
            setVoteStatus('upvote');
            setStatusUpvote(true);
          }
        }

        if (item.latest_reactions.downvotes !== undefined) {
          const downvotes = item.latest_reactions.downvotes.filter(
            (vote) => vote.user_id === selfUserId
          );
          if (downvotes !== undefined) {
            setVoteStatus('downvote');
            setStatusDowvote(true);
          }
        }
      }
    };
    validationStatusVote();
  }, [item, selfUserId]);

  React.useEffect(() => {
    const initialVote = () => {
      const c = getCountVote(item);
      setTotalVote(c);
    };
    initialVote();
  }, [item]);

  React.useEffect(() => {
    const initial = () => {
      const reactionCount = item?.reaction_counts || {};
      if (JSON.stringify(reactionCount) !== '{}') {
        const comment = reactionCount?.comment;
        if (comment !== undefined) {
          if (comment > 0) {
            setReaction(true);
            setPreviewComment(item?.latest_reactions?.comment[0]);
          }
        }
      }
    };
    initial();
  }, [item]);

  return (
    <View>
      <View style={styles.wrapperFooter}>
        <Footer
          key={itemId}
          totalComment={getCountCommentWithChild(item)}
          totalVote={totalVote}
          statusVote={voteStatus}
          onPressBlock={onPressBlock}
          onPressShare={() => onPressShare(item)}
          onPressComment={() => onPressComment(item)}
          onPressDownVote={() => {
            setStatusDowvote((prevParam) => {
              const prev = !prevParam;
              onPressDownVote({
                activity_id: item.id,
                status: prev,
                feed_group: 'domain',
                domain: item.domain.name
              });
              if (prev) {
                setVoteStatus('downvote');
                if (statusUpvote === true) {
                  setTotalVote((p) => p - 2);
                } else {
                  setTotalVote((p) => p - 1);
                }
                setStatusUpvote(false);
              } else {
                setVoteStatus('none');
                setTotalVote((p) => p + 1);
              }
              return prev;
            });
          }}
          onPressUpvote={() => {
            setStatusUpvote((prevParam) => {
              const prev = !prevParam;
              onPressUpvote({
                activity_id: item.id,
                status: prev,
                feed_group: 'domain',
                domain: item.domain.name
              });
              if (prev) {
                setVoteStatus('upvote');
                if (statusDownvote === true) {
                  setTotalVote((p) => p + 2);
                } else {
                  setTotalVote((p) => p + 1);
                }
                setStatusDowvote(false);
              } else {
                setVoteStatus('none');
                setTotalVote((p) => p - 1);
              }
              return prev;
            });
          }}
        />
      </View>
      {isReaction && (
        <View style={styles.reactionContainer}>
          <PreviewComment
            user={previewComment.user}
            comment={previewComment.data.text}
            image={previewComment.user.data.profile_pic_url}
            time={previewComment.created_at}
            totalComment={item.latest_reactions.comment.length - 1}
            onPress={() => onPressComment(item)}
          />
          <Gap height={16} />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  wrapperFooter: {
    paddingHorizontal: 8,
    height: 52,
    borderBottomColor: COLORS.gray1,
    borderBottomWidth: 1
  },
  reactionContainer: {
    zIndex: 1000
  }
});

export default LinkContextScreenFooter;
