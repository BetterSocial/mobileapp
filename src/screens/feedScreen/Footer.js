import React from 'react';
import {
  View,
  StyleSheet,
  Text,
  Platform,
  Dimensions,
  TouchableOpacity,
} from 'react-native';

import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

import MemoIc_block_inactive from '../../assets/block/Ic_block_inactive';
import MemoIc_arrow_upvote_off from '../../assets/arrow/Ic_arrow_upvote_off';
import MemoIc_arrow_down_vote_off from '../../assets/arrow/Ic_arrow_down_vote_off';
import MemoIc_share from '../../assets/icons/Ic_share';
import MemoIc_comment from '../../assets/icons/Ic_comment';

const {width: screenWidth} = Dimensions.get('window');

const Footer = ({
  onPressShare,
  onPressComment,
  onPressBlock,
  onPressUpvote,
  onPressDownVote,
  item,
  totalVote = 0,
  totalComment = 0,
}) => {
  return (
    <View style={{...styles.rowSpaceBeetwen, marginBottom: 8}}>
      <View style={{...styles.rowSpaceBeetwen, width: screenWidth * 0.25}}>
        <TouchableOpacity onPress={onPressShare}>
          <MemoIc_share height={20} width={20} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressComment}>
          <MemoIc_comment height={20} width={20} />
        </TouchableOpacity>
        <Text style={styles.textCount}>{totalComment}</Text>
      </View>
      <View style={{...styles.rowSpaceBeetwen, width: screenWidth * 0.3}}>
        <TouchableOpacity onPress={() => onPressBlock(item)}>
          <MemoIc_block_inactive height={18} width={18} />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => onPressDownVote(item)}>
          <MemoIc_arrow_down_vote_off width={18} height={18} />
        </TouchableOpacity>
        {totalVote > 0 && (
          <Text
            style={[
              styles.textCount,
              {
                color:
                  totalVote > 0
                    ? '#00ADB5'
                    : totalVote < 0
                    ? '#FF2E63'
                    : 'black',
              },
            ]}>
            {totalVote}
          </Text>
        )}
        <TouchableOpacity onPress={() => onPressUpvote(item)}>
          <MemoIc_arrow_upvote_off width={18} height={18} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Footer;

const styles = StyleSheet.create({
  rowSpaceBeetwen: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  textCount: {
    fontFamily: fonts.inter[400],
    fontSize: 18,
    lineHeight: 24,
    color: '#C4C4C4',
  },
});
