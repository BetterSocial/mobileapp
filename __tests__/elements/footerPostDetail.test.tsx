import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Footer from '../../src/elements/PostDetail/Footer';

configure({adapter: new Adapter()});

describe('component Footer', () => {
  it('Footer snapshot', () => {
    const component = shallow(<Footer onBlock={() => {}} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Footer renders correctly', () => {
    const tree = renderer.create(<Footer onBlock={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
