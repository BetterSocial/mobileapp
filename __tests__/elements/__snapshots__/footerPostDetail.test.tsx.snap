// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Footer Footer renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginHorizontal": 22,
      "marginTop": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": 10,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="ios-share-social-outline"
      size={17}
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": 10,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="block"
      size={17}
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginHorizontal": 10,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="chat"
      size={17}
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 20,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="arrow-up-bold-outline"
      size={17}
    />
  </View>
  <Text
    style={
      Object {
        "color": "#C4C4C4",
        "fontFamily": "Inter-Regular",
        "fontSize": 14,
        "marginHorizontal": 13,
      }
    }
  >
    0
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="arrow-down-bold-outline"
      size={17}
    />
  </View>
</View>
`;

exports[`component Footer Footer snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginHorizontal": 22,
      "marginTop": 20,
    }
  }
>
  <ForwardRef
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="ios-share-social-outline"
      size={17}
    />
  </ForwardRef>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="block"
      size={17}
    />
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="chat"
      size={17}
    />
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    <Icon
      color="#C4C4C4"
      name="arrow-up-bold-outline"
      size={17}
    />
  </ForwardRef>
  <Text
    style={
      Object {
        "color": "#C4C4C4",
        "fontFamily": "Inter-Regular",
        "fontSize": 14,
        "marginHorizontal": 13,
      }
    }
  >
    0
  </Text>
  <ForwardRef>
    <Icon
      color="#C4C4C4"
      name="arrow-down-bold-outline"
      size={17}
    />
  </ForwardRef>
</View>
`;
